packages/components/src/styling/animations.ts

```diff
@@ -0,0 +1,40 @@
+import {
+    useSharedValue,
+    withTiming,
+    useAnimatedStyle,
+    interpolateColor,
+} from "react-native-reanimated";
+import { useToken } from "../hooks/useToken";
+
+export const useFadeAnimation = () => {
+    const token = useToken();
+    const fadeAnimation = useSharedValue(0);
+
+    const handlePressIn = () => {
+        fadeAnimation.value = withTiming(1, {
```

Comment:
I think the fading should only be applied on the pressOut. This seems to also animate the button on press in, which feels a little bit sluggish since there is a delay on every press